temporal-sdk/src/test/java/io/temporal/workflow/TerminatedWorkflowQueryTest.java

```diff
@@ -65,6 +74,58 @@ public void testShouldReturnQueryResultAfterWorkflowTimeout() {
     Assert.assertEquals("started", client.getTrace().get(0));
   }
 
+  @Test
+  public void testShouldReturnQueryResultAfterTerminatedWorkflow() {
+    // Get a Workflow service client which can be used to start, Signal, and Query Workflow
+    // Executions.
+    TestWorkflowEnvironment testEnv =
+        TestWorkflowEnvironment.newInstance(
+            TestEnvironmentOptions.newBuilder()
+                .setWorkerFactoryOptions(WorkerFactoryOptions.newBuilder().build())
+                .build());
```

Comment:
I think you either use testWorkflowRule or TestWorkflowEnvironment. This unit test mixes them both which is really weird.